02-Logging/01-Displaying-Status-Information.ipynb

```diff
@@ -0,0 +1,353 @@
+{
```

Comment:
<p>I agree with your comment below that we don't need an exhaustive feature description here, however I think it would be useful to indicate that users can automatically log all quantities associated with a Loggable object by not specifying the <code>quantities</code> parameter since that's likely to be a common default use case.</p><p> </p><p>From an API perspective, what's the reason for <code>logger.add</code>  to return these tuples? Is it useful for the user to know what namespace these things will be logged into, and is there no other way for the user to query for that information?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/glotzerlab/hoomd-examples/pull/18/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
<p>The next tutorial section demonstrates using <code>logger.add</code>  to add all quantities. I don't think it is too much to also mention that here, so I'll add it.</p><p> </p><p>@b-butler will have to comment on the return value of <code>logger.add</code>.  I also don't see a particular need for the return value.</p>

Reply:
<p>When multiple of the same type are logged for instance two LJ potentials, we would append the <code>LJ</code>  part of the namespace with <code>LJ_1</code> , <code>LJ_2</code>,  and so on. If a user logs only some quantities in one of the objects and all on the other, the same object may use different suffixes (or none at all), since we only search for a unique namespace. We return all the namespaces used so that a user could store them to later retrieve from a logger backend.</p>

Reply:
If the user does not store the return of `logger.add`, is there any way to disambiguate between the two sets of quantities (i.e. to tell which logged quantity corresponds to which potential) after the fact?

Reply:
No, not currently, and certainly not once we actually log the values.

Reply:
<p>It might be worthwhile to provide explicit user-provided disambiguation suffixes for cases like this.</p>

Reply:
Yes, that seems problematic if the information is lost after the script has completed. Otherwise users have to know to save the names returned from `logger.add` so that the information persists past the run, which doesn't seem desirable.

Reply:
We can support user provided suffixes, but the problem would still be there if they don't use that argument. I don't think there is a way to not error in logging multiple instances of the same class without doing something like we already do.

Reply:
Currently (unless this has changed, but I don't think it could have changed without substantial C++ modifications) almost every C++ object in HOOMD has a name that is provided to it on construction. I think that has been removed from the Python API for constructing pair potentials, is that correct? If so, this discussion makes me think that we should add it back, and require users to provide names if they want to know which LJ log corresponds to which object (for example). I don't see a way around this problem that would be more transparent than simply asking the user to specify a name for objects if they want to have multiple and log them. An error up front would be much better than having them run the simulation only to realize they don't know which energy is `LJ_1` and which is `LJ_2`.

Reply:
That is only true for cases where it matters which is which, and that is not always the case. I don't think a user should have to provide a name to create an object, and if they don't, I don't think we should error on them logging the object. We could provide the name option at the object level, but I wouldn't want to error if we needed to iterate over namespaces. Also, allowing names means we would have to check for uniqueness or allow duplicate name where the logger would add a suffix again leading to `user_name_1` and so on.

Reply:
What I'm suggesting is adding logging to the list of cases where it matters which object is which. In other words, a user is free to create two separate unnamed instances of `Pair.LJ` and run a simulation. However, if a user attempts to _log_ both of them, then raise an error and require the user to provide a unique name. This would remove the need for arbitrary suffixes, and instead put the onus on the user to specifiy which object is which in a way that ensures that they also know how to parse the log output based on what names they used in their script.

Reply:
The C++ naming code is still there, but I do plan to remove it. I don't think every object should have a name, that decouples the name from where it gets logged. I think the user should be able to provide a name suffix in the call to logger.add: 
```logger.add(obj1, suffix="type_A")
logger.add(obj2, suffix="type_B")
```
We could consider adding this suffix to the tuple rather than as an underscore.

I cannot imagine a case where a user wants to log two values with the same name and also does not need to be able to identify them. I think it would be more natural for users to get an error message when they attempt this - that clearly lets them know that the need to add the name suffixes.

Reply:
I agree, I don't think every object needs a name. I was suggesting to repurpose that API, but a more dedicated API for the purpose of logging would definitely be a cleaner solution. Your second paragraph is my primary point: any time a user logs multiple of the same object, I can't imagine any reason they wouldn't need to differentiate them, and since a user-provided name of some sort is the easiest option we should make it an error if one isn't provided.